tests/conftest.py

```diff
@@ -974,24 +973,58 @@ def disable_editor():
 
 
 @pytest.fixture()
-def postgis_db():
+def postgres_db():
     """
-    Using docker, you can run a PostGIS test - such as test_postgis_import - as follows:
+    Using docker, you can run all PostgreSQL /  PostGIS tests - such as test_postgis_import - as follows:
         docker run -it --rm -d -p 15432:5432 -e POSTGRES_HOST_AUTH_METHOD=trust postgis/postgis
-        KART_POSTGRES_URL='postgresql://postgres:@localhost:15432/postgres' pytest -k postgis --pdb -vvs -n 0
+        KART_POSTGRES_URL='postgresql://postgres:@localhost:15432/postgres' pytest -k "postgres OR postgis" --pdb -vvs
     """
     if "KART_POSTGRES_URL" not in os.environ:
         raise pytest.skip(
-            "PostGIS tests require configuration - read docstring at conftest.postgis_db"
+            "Postgres tests require configuration - read docstring at conftest.postgres_db"
         )
     engine = Db_Postgis.create_engine(os.environ["KART_POSTGRES_URL"])
-    with engine.connect() as conn:
+    yield engine
+
+
+@pytest.fixture()
+def postgis_db(postgres_db):
+    # See postgres_db for help on running PostGIS tests
+    # engine = Db_Postgis.create_engine(os.environ["KART_POSTGRES_URL"])
+    with postgres_db.connect() as conn:
+        # with engine.connect() as conn:
         # test connection and postgis support
         try:
             conn.execute("SELECT postgis_version();")
         except sqlalchemy.exc.DBAPIError:
             raise pytest.skip("Requires PostGIS")
-    yield engine
+    yield postgres_db
+
+
+@pytest.fixture()
+def no_postgis_db(postgres_db):
+    # See postgres_db for help on running no_postgis_db tests.
+    # Check if the user has superuser access to the database:
+    is_superuser = False
+    with postgres_db.connect() as conn:
+        is_superuser = conn.scalar(
+            "SELECT rolsuper FROM pg_roles WHERE rolname = current_user;"
+        )
+
+    if not is_superuser:
+        raise SystemError("Requires superuser access to the database")
+
+    try:
+        with postgres_db.begin() as transaction:
+            transaction.execute("DROP EXTENSION IF EXISTS postgis;")
```

Comment:
i think we need this fixture to `yield` inside this transaction, so that the test runs in a transaction where postgis extension has been dropped, but when we rollback (for other tests) it gets un-dropped again.

The line below (which always causes an error) causes the transaction to roll back, but we'd probably be better off with an explicit rollback - does the connection object have a rollback method?